iib/workers/tasks/opm_operations.py

```diff
@@ -190,3 +191,201 @@ def _serve_cmd_at_port(serve_cmd, cwd, port, max_tries, wait_time):
         rpc_proc.kill()
 
     raise IIBError(f'Index registry has not been initialized after {max_tries} tries')
+
+
+def opm_migrate(index_db, base_dir):
+    """
+    Migrate SQLite database to File-Based catalog using opm command.
+
+    :param str index_db: path to SQLite index.db which should migrated to FBC.
+    :param str base_dir: base directory to which the database file should be saved.
+    :return: Returns path to directory containing file-based catalog.
+    :rtype: str
+    """
+    from iib.workers.tasks.utils import run_cmd
+
+    fbc_dir_path = os.path.join(base_dir, 'catalog')
+
+    if os.path.exists(fbc_dir_path):
+        shutil.rmtree(fbc_dir_path)
+
+    cmd = ['opm', 'migrate', index_db, fbc_dir_path]
+
+    run_cmd(cmd, {'cwd': base_dir}, exc_msg='Failed to migrate index.db to file-based catalog')
+    log.info("Migration to file-based catalog was completed.")
+    return fbc_dir_path
+
+
+def omp_generate_dockerfile(fbc_dir, base_dir, index_db, binary_image=None, dockerfile_name=None):
+    """
+    Generate Dockerfile using opm command and adding index.db to hidden location.
+
+    :param str fbc_dir: directory containing file-based catalog (JSON or YAML files).
+    :param str base_dir: base directory to which the database file should be saved.
+    :param str index_db: path to SQLite index.db which should be put to hidden location in container
+    :param str binary_image: pull spec of binary image in which to build catalog.
+    :param str dockerfile_name: name of generated Dockerfile.
+    :return: Returns path to generated Dockerfile
+    :raises: IIBError when Dockerfile was not generated
+    :rtype: str
+    """
+    from iib.workers.tasks.utils import run_cmd
+
+    cmd = [
+        'opm',
+        'alpha',
+        'generate',
+        'dockerfile',
+        os.path.abspath(fbc_dir),
+    ]
+
+    if binary_image:
```

Comment:
Sorry for my lack of knowledge, but why is `binary_image` optional? And what's the difference between providing one and not providing one.

Reply:
Good question. I made it optional because it is optional for `opm` it self. 
The difference is what base image will be used in Dockerfile (specifying `FROM image:latest`). 
If binary_image is not set it defaults to `quay.io/operator-framework/opm:latest`.

And since you are asking I guess we do not want to have this behavior in IIB because we want to use our binary image as a base image, right? It would be safe to make it mandatory.

Reply:
@yashvardhannanavati WDYT?

Reply:
Agree, its mandatory now.